tools/topology/topology2/cavs/CMakeLists.txt

```diff
@@ -44,6 +44,12 @@ PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-cavs-tgl-nocodec.bin"
 # SSP topology for MTL
 "cavs-nocodec\;sof-mtl-nocodec\;PLATFORM=mtl,NUM_DMICS=2,\
 PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-ace-mtl-nocodec.bin"
+
+# SSP topology for MTL
+"cavs-rt5682\;sof-mtl-max98357a-rt5682\;PLATFORM=mtl,\
+NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,\
+NUM_HDMIS=4,PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-ace-mtl-rt5682.bin"
```

Comment:
I am still very very confused with the NHLT naming. We need separate SSP blobs for rt5682 and max98357a, and there's no mention on what the dmic configuration is. That's a recipe for confusion and possible disaster when reusing stuff.

Reply:
Do you mean NHLT_BIN? true, any established NHLT naming convention that I can follow?

Reply:
@plbossart IIRC, the NHLT_BIN is the hnlt blob of this topology. The purpose is that we can analyze the nhlt bin file if we want to debug the nhlt blob. So the nhlt bin is associated to the .tplg file. Maybe we can always use the same name as the .tplg file? Like sof-mtl-max98357a-rt5682.tplg and nhlt-sof-mtl-max98357a-rt5682.bin.
@ranj063 @juimonen Please correct me if I was wrong.

Reply:
Yes the .bin is debug feature to extract nhlt blob from the process (so you don't have to dig it out from the topology binary). Not sure should we have some general DEBUG_TOPOLOGY flag in CMake. I don't personally mind it being there, but others might when the amount of topologies grows.

Reply:
This NHLT needs to be referenced by source file name or by coding the config in the topology (when we are not reading from ACPI).